pkg/cri/resource-manager/cache/cache.go

```diff
@@ -1110,6 +1128,47 @@ func (cch *cache) setEntry(key string, ptr, obj interface{}) error {
 	return nil
 }
 
+// checkFilePermissions checks permission bits of an existing file.
+func (cch *cache) checkFilePermissions(path string, perm, rejected os.FileMode, what string) error {
+	if info, err := os.Stat(path); err == nil && !info.IsDir() {
+		expected := perm & os.ModePerm // deliberately ignore sticky bit in comparison
+		existing := info.Mode().Perm() & os.ModePerm
+		if rejected != 0 && (existing&rejected) != 0 {
+			return cacheError("existing %s file %q has disallowed permissions set: %v",
+				what, path, existing&rejected)
+		}
+		if existing != expected && (existing|expected) != expected {
+			cch.Warn("existing %s file %q has less strict permissions %v than expected %v",
+				what, path, existing, expected)
+		}
+	}
+
+	return nil
+}
+
+// mkdirAll creates a directory if it does not exist, checks/warns about its permissions if it does.
+func (cch *cache) mkdirAll(path string, perm, rejected os.FileMode, what string) error {
+	if info, err := os.Stat(path); err == nil && info.IsDir() {
+		expected := perm & os.ModePerm // deliberately ignore sticky bit in comparison
+		existing := info.Mode().Perm() & os.ModePerm
+		if rejected != 0 && (existing&rejected) != 0 {
+			return cacheError("existing %s directory %q has disallowed permissions set: %v",
+				what, path, existing&rejected)
+		}
+		if existing != expected && (existing|expected) != expected {
+			cch.Warn("existing %s directory %q has less strict permissions %v than expected %v",
+				what, path, existing, expected)
+		}
```

Comment:
Is this duplicated from `checkFilePermissions()` if so, could it be split out in a separate helper function?

Reply:
Split out/combined permission checks.